src/sqs.ts

```diff
@@ -92,32 +107,68 @@ export class SQSMessageHandler<Message, Context> {
       // 2. Process all the records.
       context.logger.info({ event }, 'Processing SQS topic message');
 
-      await processWithOrdering(
-        {
-          items: event.Records,
-          // If there is not a MessageGroupId, then we don't care about
-          // the ordering for the event. We can just generate a UUID for the
-          // ordering key.
-          orderBy: (record) => record.attributes.MessageGroupId ?? uuid(),
-          concurrency: this.config.concurrency ?? 5,
-          stopOnError: false,
-        },
-        async (record) => {
-          const messageLogger = context.logger.child({
-            messageId: record.messageId,
-          });
-
-          const parsedMessage = this.config.parseMessage(record.body);
-
-          for (const action of this.messageActions) {
-            await action({ ...context, logger: messageLogger }, parsedMessage);
-          }
-
-          messageLogger.info('Successfully processed message');
-        },
-      );
-
-      context.logger.info('Succesfully processed all messages');
+      const { hasUnprocessedRecords, unprocessedRecords } =
+        await processWithOrdering(
+          {
+            items: event.Records,
+            // If there is not a MessageGroupId, then we don't care about
+            // the ordering for the event. We can just generate a UUID for the
+            // ordering key.
+            orderBy: (record) => record.attributes.MessageGroupId ?? uuid(),
+            concurrency: this.config.concurrency ?? 5,
+            rejectOnError: false,
+          },
+          async (record) => {
+            const messageLogger = context.logger.child({
+              messageId: record.messageId,
+            });
+
+            const parsedMessage = this.config.parseMessage(record.body);
+
+            for (const action of this.messageActions) {
+              await action(
+                { ...context, logger: messageLogger },
+                parsedMessage,
+              );
+            }
+
+            messageLogger.info('Successfully processed message');
+          },
+        );
+
+      if (hasUnprocessedRecords) {
+        context.logger.warn(
+          { unprocessedRecords },
+          'Failed to process all messages',
+        );
+      } else {
+        context.logger.info('Succesfully processed all messages');
+      }
```

Comment:
Since the details of individual failures will be logged below, I think we could just skip this log + simplify like so:
```suggestion
      if (!hasUnprocessedRecords) {
        context.logger.info('Succesfully processed all messages');
      }
```